ssl/quic/quic_channel.c

```diff
@@ -2000,9 +2004,11 @@ static void ch_rx_handle_packet(QUIC_CHANNEL *ch)
             /* Malformed retry packet, ignore. */
             return;
 
-        ch_retry(ch, ch->qrx_pkt->hdr->data,
-                 ch->qrx_pkt->hdr->len - QUIC_RETRY_INTEGRITY_TAG_LEN,
-                 &ch->qrx_pkt->hdr->src_conn_id);
+        if (!ch_retry(ch, ch->qrx_pkt->hdr->data,
+                      ch->qrx_pkt->hdr->len - QUIC_RETRY_INTEGRITY_TAG_LEN,
+                      &ch->qrx_pkt->hdr->src_conn_id))
+            ossl_quic_channel_raise_protocol_error(ch, QUIC_ERR_INTERNAL_ERROR,
```

Comment:
Failure of `ch_retry` is not a protocol error. See the first comment in ch_retry &mdash; if a packet isn't handled we just drop it. As this is currently written a malicious third party could force connection termination by sending a packet with the wrong retry SCID...

Reply:
@hlandau Really? The case you talk about returns 1 from ch_retry(). If an error is returned from ch_retry(), it is an internal error and IMO that should be handled.

Reply:
See `ch_retry`:

```c
    /*
     * RFC 9000 s. 17.2.5.1: "A client MUST discard a Retry packet that contains
     * a SCID field that is identical to the DCID field of its initial packet."
     */
    if (ossl_quic_conn_id_eq(&ch->init_dcid, retry_scid))
        return 0;
```

You would at least need to change this to `return 1`. The other cases are probably OK to handle as internal errors.

Reply:
Heh, I see return 1 in the branch there. I need to rebase it as it was changed later to return 0.